File: VOWELS/two_vowels.js
import fs from 'fs';

function removePunctuation(word) {
    return word.replace(/[^a-zA-Z]/g, '');  
}

function containsConsecutiveVowels(word) {
    const vowelPattern = /[aeiouAEIOU]{2}/;  
    return vowelPattern.test(word);
}

function findWordsWithConsecutiveVowels(filename) {
    fs.readFile(filename, 'utf8', (err, data) => {
        if (err) {
            console.error(`Error opening file ${filename}:`, err);
            return;
        }

        const words = data.split(/\s+/); 
        words.forEach(word => {
            const cleanWord = removePunctuation(word);
            if (containsConsecutiveVowels(cleanWord)) {
                console.log(cleanWord);  
            }
        });
    });
}

findWordsWithConsecutiveVowels('dictionary.txt');

File: VOWELS/two_vowels.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <regex>
#include <cctype>

using namespace std;

string removePunctuation(const string &word) {
    string cleanWord;
    for (char ch : word) {
        if (isalpha(ch)) {
            cleanWord += ch;
        }
    }
    return cleanWord;
}

bool containsConsecutiveVowels(const string &word) {
    regex vowelPattern("[aeiouAEIOU]{2}");  
    return regex_search(word, vowelPattern);
}

void findWordsWithConsecutiveVowels(const string &filename) {
    ifstream file(filename);
    if (!file) {
        cout << "Error opening file " << filename << endl;
        return;
    }

    string word;
    while (file >> word) {
        string cleanWord = removePunctuation(word);
        if (containsConsecutiveVowels(cleanWord)) {
            cout << cleanWord << endl;  
        }
    }
    file.close();

}

int main() {
    findWordsWithConsecutiveVowels("dictionary.txt");
    return 0;
}
